Bug: make playground-up fails — bun.lock out of sync with dashboard/package.json

## What to build

`make playground-up` fails during the dashboard playground image build. The Dockerfile runs `bun install --frozen-lockfile`, but the committed `bun.lock` no longer matches `dashboard/package.json`, so bun aborts:

```
[1/2] STEP 4/6: RUN bun install --frozen-lockfile
bun install v1.3.12
Resolving dependencies
Resolved, downloaded and extracted [42]
error: lockfile had changes, but lockfile is frozen
note: try re-running without --frozen-lockfile and commit the updated lockfile
building at STEP "RUN bun install --frozen-lockfile": while running runtime: exit status 1
make: *** [Makefile:67: playground-up] Error 1
```

Reproduced on clean `main` (HEAD `dc6c862`, working tree clean). The stub-mcp and task-service playground images build fine; the failure is isolated to the dashboard stage. `make playground-down` works correctly.

Regenerate and commit `bun.lock` so it is in sync with the manifest and `--frozen-lockfile` passes.

## Acceptance criteria

- [ ] `bun.lock` is in sync with `dashboard/package.json` (a fresh `bun install` produces no lockfile changes)
- [ ] `make playground-up` completes the dashboard image build without the frozen-lockfile error
- [ ] `make playground-up` brings the full playground stack up; `make playground-down` tears it down cleanly

## Blocked by

None - can start immediately.

--- a/dashboard/Dockerfile
+++ b/dashboard/Dockerfile
@@ -1,14 +1,15 @@
 FROM oven/bun:1.3.12-alpine AS builder
 WORKDIR /app
-COPY dashboard/package.json bun.lock ./
-RUN bun install --frozen-lockfile
-COPY dashboard/ .
-RUN bun run build
+COPY package.json bun.lock ./
+COPY dashboard/package.json ./dashboard/
+COPY dashboard/ ./dashboard/
+RUN cd dashboard && bun install --frozen-lockfile
+RUN cd dashboard && node_modules/.bin/react-router build
 
 FROM oven/bun:1.3.12-alpine
 WORKDIR /app
-COPY --from=builder /app/build ./build
-COPY --from=builder /app/node_modules ./node_modules
-COPY --from=builder /app/package.json ./
+COPY --from=builder /app/dashboard/build ./build
+COPY --from=builder /app/dashboard/node_modules ./node_modules
+COPY --from=builder /app/dashboard/package.json ./
 EXPOSE 3000
 CMD ["bun", "build/server/index.js"]

--- a/dashboard/app/components/status-badge.tsx
+++ b/dashboard/app/components/status-badge.tsx
@@ -1,0 +1,1 @@
+export { StatusBadge } from "./StatusBadge";

--- a/dashboard/app/routes.ts
+++ b/dashboard/app/routes.ts
@@ -1,0 +1,11 @@
+import { type RouteConfig, index, route } from "@react-router/dev/routes";
+
+export default [
+  index("routes/_index.tsx"),
+  route("handlers", "routes/handlers._index.tsx"),
+  route("handlers/:name", "routes/handlers.$name.tsx"),
+  route("sse/tasks", "routes/sse.tasks.ts"),
+  route("tasks", "routes/tasks._index.tsx"),
+  route("tasks/:id", "routes/tasks.$id.tsx"),
+  route("tasks/:id/lineage", "routes/tasks.$id.lineage.tsx"),
+] satisfies RouteConfig;

--- a/dashboard/app/routes/tasks._index.tsx
+++ b/dashboard/app/routes/tasks._index.tsx
@@ -6,7 +6,6 @@ import {
   useNavigate,
   useRevalidator,
 } from "react-router";
-import { listTasks } from "~/lib/api.server";
 import type { Task, TaskList } from "~/lib/api.server";
 import { shortId, canTimeout, canRetry } from "~/lib/utils";
 import { StatusBadge } from "~/components/StatusBadge";
@@ -182,14 +181,15 @@ export default function TasksIndex() {
       const status = params.get("status") ?? undefined;
       const since = params.get("since") ?? undefined;
 
-      const data = await listTasks({
-        handler: handler ?? undefined,
-        status: status ?? undefined,
-        since: since ?? undefined,
-        sort: "received_at:desc",
-        limit: 50,
-        cursor: nextCursor,
-      });
+      const qs = new URLSearchParams();
+      if (handler) qs.set("handler", handler);
+      if (status) qs.set("status", status);
+      if (since) qs.set("since", since);
+      qs.set("sort", "received_at:desc");
+      qs.set("limit", "50");
+      qs.set("cursor", nextCursor);
+      const res = await fetch(`/tasks?${qs.toString()}`);
+      const { data } = (await res.json()) as { data: TaskList };
 
       // Merge into map (de-dup by ID)
       const map = taskMapRef.current;

--- a/dashboard/vite.config.ts
+++ b/dashboard/vite.config.ts
@@ -1,7 +1,13 @@
 import { reactRouter } from "@react-router/dev/vite";
 import tailwindcss from "@tailwindcss/vite";
 import { defineConfig } from "vite";
+import { resolve } from "path";
 
 export default defineConfig({
   plugins: [tailwindcss(), reactRouter()],
+  resolve: {
+    alias: {
+      "~": resolve(__dirname, "app"),
+    },
+  },
 });

